### PR TITLE
Add UpdateSchemaLockCommand and utility classes

### PR DESCRIPTION
**Pull request by q-hung**
Changes proposed in this pull request:
* Add UpdateSchemaLockCommand and utility classes
* Tidy up code in GraphQLServiceProvider
* Unit tests
* .travis.yml
* Build status badge, coverall status badge

### CODE REVIEW

File: src/Helpers/JsonDecoder.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
```
Comment (Jalle19) [inline src/Helpers/JsonDecoder.php]: These two classes are available in https://packagist.org/packages/digiaonline/json-helpers nowadays

**Review by Jalle19 — CHANGES_REQUESTED** [review]

**Comment by cniska** [comment]
@hungneox can you please rename this PR to reflect the changes a bit better?

File: composer.json [hunk 2]
```diff
@@ -3,11 +3,14 @@
     "description": "GraphQL module for the Laravel and Lumen PHP frameworks.",
     "type": "library",
     "require": {
-        "youshido/graphql": "^1.5"
```
Comment (Jalle19) [inline composer.json]: Could you add `"php": ">=7.1"` as a dependency too?

File: src/ArrayConnection.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (cniska) [inline src/ArrayConnection.php]: I'm not sure that we need this as a part of this library as it's kind of a hack?

File: src/Helpers/JsonEncoder.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
```
Comment (cniska) [inline src/Helpers/JsonEncoder.php:1]: Shouldn't we use the library @Jalle19 made for this?

File: src/GraphQLServiceProvider.php [hunk 5]
```diff
@@ -20,8 +20,20 @@ public function register()
     {
         // Configure
         $this->app->configure('graphql');
+        // Load graphiql view
         $this->loadViewsFrom(dirname(__DIR__) . '/resources/views', 'graphql');
```
Comment (cniska) [inline src/GraphQLServiceProvider.php]: If we want to support both Lumen and Laravel we need an if-block here.

Comment (cniska) [reply src/GraphQLServiceProvider.php]: Like so: https://github.com/digiaonline/lumen-cors/blob/master/src/CorsServiceProvider.php#L15

**Review by cniska — CHANGES_REQUESTED** [review]
Looks good, added some comments that require some changes.

File: src/Helpers/JsonEncoder.php (hunk 4 above)
Comment (Jalle19) [reply src/Helpers/JsonEncoder.php:1]: @crisu83 we are, but we need to override the encoder to get rid of those unwanted characters

File: src/GraphQLServiceProvider.php (hunk 5 above)
Comment (Jalle19) [reply src/GraphQLServiceProvider.php]: I think we should focus mostly on Lumen for now

Comment (cniska) [reply src/GraphQLServiceProvider.php]: This is a standard way of adding support for both frameworks. I don't understand why we would do it differently in this project?

File: src/Helpers/JsonEncoder.php (hunk 4 above)
Comment (cniska) [reply src/Helpers/JsonEncoder.php:1]: Fair enough.

File: src/Http/GraphQLController.php [hunk 6]
```diff
@@ -28,18 +31,35 @@ public function __construct(GraphQLService $graphqlService)
...
 
         $query     = $request->get('query');
-        $variables = $request->has('variables') ? $request->get('variables') : [];
+        $variables = $request->get('variables', []);
 
         $responseData = $processor->processPayload($query, $variables)->getResponseData();
 
-        return response()->json($this->responseDataToJson($responseData));
+        if (isset($responseData['exceptions'])) {
+            $request->attributes->set($this->getErrorAttribute(), new GraphQLError(
```
Comment (q-hung) [inline src/Http/GraphQLController.php:47]: @crisu83 @Jalle19 I just made the error attribute configurable here.

Comment (Jalle19) [reply src/Http/GraphQLController.php:47]: +1

File: composer.json [hunk 7]
```diff
@@ -3,11 +3,16 @@
...
+        "youshido/graphql": "^1.5",
+        "jalle19/laravel-lost-interfaces": "^1.0",
+        "digiaonline/json-helpers": "^1.0",
+        "php": ">=7.1"
```
Comment (Jalle19) [inline composer.json]: Can you add `"sort-packages": true` to the `config` section?

Comment (Jalle19) [reply composer.json]: Once you've added sort-packages, can you require https://packagist.org/packages/illuminate/contracts as a dependency? We use for some excpetions and what not, and requiring a new package will also cause Composer to sort the list of packages once.

Comment (Jalle19) [reply composer.json]: Actually add `illuminate/console` and `illuminate/support` too.

**Review by Jalle19 — CHANGES_REQUESTED** [review]

**Comment by Jalle19** [comment]
@hungneox I enabled Travis CI for this project so the next time you push something it should build and also upload coverage statistics to Coveralls.

**Review by Jalle19 — APPROVED** [review]
Let's get this in, it's getting so long that it's getting hard to review anyway. At least the tests pass etc.

**Comment by cniska** [comment]
Awesomesauce!

**Comment by cniska** [comment]
Oops, old habit.

**Comment by cniska** [comment]
@hungneox why wasn't this done in a feature branch?

**Comment by q-hung** [comment]
@crisu83 Because there are so many small things in this PR. IMO, I thought that if we have a stable branch and a develop branch it helps releasing easier (not sure :D).

**Comment by q-hung** [comment]
Close #2

**Comment by hugovk** [comment]
About the develop branch:

This project's [CONTRIBUTING file](https://github.com/digiaonline/lumen-graphql/blob/master/.github/CONTRIBUTING.md) says "Always create pull requests to the *develop* branch", but [last March](https://nordsoftware.slack.com/archives/C0DH80HNZ/p1488379290000374) we decided to follow something closer to [GitHub Flow](https://guides.github.com/introduction/flow/) for new projects (discussion begins around [here](https://nordsoftware.slack.com/archives/C0DH80HNZ/p1488374782000314)). 

So I'll delete this develop branch and update CONTRIBUTING.

**Comment by Jalle19** [comment]
@hungneox please tell me you have the branch locally still?

**Comment by Jalle19** [comment]
WTF never mind

**Pull request merged**